c++/nda/basic_array.hpp

```diff
@@ -255,13 +255,13 @@ namespace nda {
 
     //------------------ Factory -------------------------
 
```

Comment:
Why this ? 
It should be deduced from the variable passed.

Reply:
The deduction will fail for cases where you provide an initializer list
```c++
 matrix_t::zeros({n, n});
```
That is why I added the default here.

Reply:
Oh, I see... nice indeed, it make the std::array on the fly.
ok.